base_layer/core/tests/mempool.rs

```diff
@@ -333,6 +333,306 @@ fn test_retrieve() {
     assert!(retrieved_txs.contains(&tx2[1]));
 }
 
+#[test]
+#[allow(clippy::identity_op)]
+fn test_zero_conf() {
+    let network = Network::LocalNet;
+    let (mut store, mut blocks, mut outputs, consensus_manager) = create_new_blockchain(network);
+    let mempool_validator = TxInputAndMaturityValidator::new(store.clone());
+    let mempool = Mempool::new(MempoolConfig::default(), Arc::new(mempool_validator));
+    let txs = vec![txn_schema!(
+        from: vec![outputs[0][0].clone()],
+        to: vec![21 * T, 11 * T, 11 * T, 16 * T]
+    )];
+    // "Mine" Block 1
+    generate_new_block(&mut store, &mut blocks, &mut outputs, txs, &consensus_manager).unwrap();
+    mempool.process_published_block(blocks[1].to_arc_block()).unwrap();
+
+    // This transaction graph will be created, containing 3 levels of zero-conf transactions, inheriting dependent
+    // outputs from multiple parents
+    //
+    // tx01   tx02   tx03   tx04    Basis transactions using mined inputs (lowest fees, increases left to right)
```

Comment:
This would be a nice future to do. Currently, we only have one test, testing this. If and when we extend this to more tests we could look into the time to build something fancier.